pkg/hubble/parser/threefour/parser.go

```diff
@@ -160,7 +170,24 @@ func (p *Parser) Decode(data []byte, decoded *pb.Flow) error {
 	// Skip decoding and truncate layers to avoid accidental re-use.
 	// See https://github.com/google/gopacket/issues/846
 	if len(data[packetOffset:]) > 0 {
-		err := p.packet.decLayer.DecodeLayers(data[packetOffset:], &p.packet.Layers)
+		var isL3Device, isIPv6 bool
+		if tn != nil && tn.IsL3Device() {
+			isL3Device = true
+		}
+		if tn != nil && tn.IsIPv6() {
+			isIPv6 = true
+		}
+
+		var err error
+		switch {
+		case !isL3Device:
+			err = p.packet.decLayerL2Dev.DecodeLayers(data[packetOffset:], &p.packet.Layers)
+		case isIPv6:
+			err = p.packet.decLayerL3DevIPv6.DecodeLayers(data[packetOffset:], &p.packet.Layers)
+		default:
+			err = p.packet.decLayerL3DevIPv4.DecodeLayers(data[packetOffset:], &p.packet.Layers)
```

Comment:
what if... it's an arp packet on wg0? Did hubble ever care about arp? If no, ignore arp. But make sure hubble won't panic or output annoying error logs.

Reply:
I'm not able to produce an ARP packet passing through `cilium_wg0`.

Anyway, the function would return an error depending on the packet: it would start decoding using, for instance, IPv4, and would receive an error such as `Unexpected error from Decode(data, *flow.Flow): Invalid (too small) IP header length (0 < 5)`.

If debug is enabled, it will be later displayed https://github.com/cilium/cilium/blob/main/pkg/hubble/observer/local_observer.go#L150.